tests/build-and-install-debs.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+# Utility script to build DEBs in a Docker container and then install them
+
+set -eo pipefail
+
+if [ "$1" == "--with-tests" ]
+then
+    RUN_TESTS=true
+    shift
+else
+    RUN_TESTS=false
+fi
+
+TAG=$1
+DOCKERFILE=$2
+
+IMAGE=cobbler:$TAG
+
+# Build container
+docker build -t "$IMAGE" -f "$DOCKERFILE" .
+
+# Build DEBs
+mkdir -p deb-build tmp
+mkdir -p deb-build/{BUILD,BUILDROOT,DEBS,SDEBS,SOURCES}
+docker run -ti -v "$PWD/deb-build:/usr/src/cobbler/deb-build" -v "$PWD/tmp:/var/tmp" "$IMAGE"
+
+# Launch container and install cobbler
+docker run -d --privileged -v /sys/fs/cgroup:/sys/fs/cgroup:ro --name cobbler -v "$PWD/deb-build:/usr/src/cobbler/deb-build" "$IMAGE" /lib/systemd/systemd --system
+
+# THIS NEEDS TO BE FIXED !! In setup.py or somewhere else.
```

Comment:
From here I had to fix things by hand for now.
For example, I don't know how to overrule the usage of `self.root` pointing at `/usr/local` in `setup.py`, L320.
How to auto enable the modules & config for Apache2; create directories in `/etc/cobbler/`.

Reply:
Okay so enabling modules of Apache2 is not the concern of the `setup.py` in my eyes. This is something an admin has to do because we don't want to mess with the webserver configuration automatically (imagine the installation fails due to some env problem and the webserver is then in an broken state). The etc directory and files are already created (at least partly) in the data files section in the `setup.py L584-613`.

Regarding `self.root`: Just use the option `--root`. The hint for this can be found in `setup.py L318`.

Reply:
This is hard, because of this line:
https://github.com/cobbler/cobbler/blob/master/Makefile#L124
and `DESTDIR=/`

Reply:
You can use the following to override this: https://www.gnu.org/software/make/manual/make.html#Overriding

Reply:
Tried really hard but I am unable to overrule `@@install_scripts@@` and `@@install_data@@`. It used to be like this:

https://github.com/cobbler/cobbler/commit/9eb1f39df9d9c430cc30fadd271a6df69f21ce43#diff-ff3ee63545d5735761cd6925c830c669

but are there ANY distros that install packages in `/usr/local` by default, except for eg. FreeBSD (ports)?

Reply:
I would actually like to see this possible because then we can use the make file to perform user installations.

Reply:
But if you can't get it to work then this is also fine for me.

Reply:
> But if you can't get it to work then this is also fine for me.

As stated, I am unable to overrule `@@install_scripts@@` and `@@install_data@@`.
I do not like my solution at all, but I can't get through the options of the `Makefile` and the `setup.py`... 😢
Looking into the `setup.py`, there must be a way to set `--install-scripts` & `--install-data` for the install option, but I don't get it.

Reply:
To share some of my confusion: in `setup.py` setting `self.root = "/usr/local"` to `/usr` does not make any difference. So I wonder, why is it (still) there?

Reply:
F*** I have no idea and this means probably that this is broken anyway. But I don't think we can remove it because there is probably something depending on it during the installation process...